File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    a = [1, 2, 3, 4, 5]
    b = [2, 1, 2, 3, 2, 4, 2, 5]
    c = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    cnt = [0, 0, 0]
    max_win = 0

    for i in range(len(answers)):
        if a[i % 5] == answers[i]:
            cnt[0] += 1

        if b[i % 8] == answers[i]:
            cnt[1] += 1

        if c[i % 10] == answers[i]:
            cnt[2] += 1

    for i in range(len(cnt)):
        if cnt[i] > max_win:
            max_win = cnt[i]
            answer = [i+1]
        elif cnt[i] == max_win:
            answer.append(i+1)

    return answer

File: 프로그래머스/lv1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int[] answer = {};
        List<Integer> answerList = new ArrayList<>();
        int[] correct = {0, 0, 0};
        int[][] pattern = {{1, 2, 3, 4, 5}, {2, 1, 2, 3, 2, 4, 2, 5}, {3, 3, 1, 1, 2, 2, 4, 4, 5, 5}};
        for (int i = 0; i < 3; i++) {
            int div = pattern[i].length;
            for (int j = 0; j < answers.length; j++) {
                if (pattern[i][j % div] == answers[j]) {
                    correct[i]++;
                }
            }
        }

        int maxScore = Arrays.stream(correct).max().getAsInt();
        for (int i = 0; i < correct.length; i++) {
            if (correct[i] >= maxScore) {
                answerList.add(i+1);
            }
        }
        answer = answerList.stream().mapToInt(i -> i).toArray();
        
        return answer;
    }
}